src/Header/OperateData.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef GROUP4_SRC_OPERATEDATA_H_
+#define GROUP4_SRC_OPERATEDATA_H_
+
+#include <string>
+#include <unordered_map>
+
+// This class is a self-defined data structure containing header and body.
+// OperateData newData = OperateData();
+// OperateData["key"] = "value";
+struct RequestHeader
+{
+    std::string Accept;
+    std::string AcceptCharset;
+    std::string AcceptEncoding;
+    std::string AcceptLanguage;
+    std::string Authorization;
+    std::string CacheControl;
+    std::string Connection;
+    std::string Cookie;
+    std::string ContentLength;
+    std::string ContentType;
+    std::string Date;
+    std::string Expect;
+    std::string From;
+    std::string Host;
+    std::string IfMatch;
+    std::string IfModifiedSince;
+    std::string IfNoneMatch;
+    std::string IfRange;
+    std::string IfUnmodifiedSince;
+    std::string MaxForwards;
+    std::string Origin;
+    std::string ProxyAuthorization;
+    std::string Range;
+    std::string Referer;
+    std::string TE;
+    std::string UserAgent;
+    std::string Upgrade;
+    std::string Via;
+    std::string Warning;
+};
+
+struct ResponseHeader
+{
+    std::string AccessControlAllowOrigin;
+    std::string AcceptPatch;
+    std::string AcceptRanges;
+    std::string Age;
+    std::string Allow;
+    std::string CacheControl;
+    std::string Connection;
+    std::string ContentDisposition;
+    std::string ContentEncoding;
+    std::string ContentLanguage;
+    std::string ContentLength;
+    std::string ContentLocation;
+    std::string ContentRang;
+    std::string ContentType;
+    std::string Dat;
+    std::string ETag;
+    std::string Expires;
+    std::string LastModified;
+    std::string Link;
+    std::string Location;
+    std::string Pragma;
+    std::string ProxyAuthenticate;
+    std::string PublicKeyPins;
+    std::string RetryAfter;
+    std::string Server;
+    std::string SetCookie;
+    std::string Status;
+    std::string StrictTransportSecurity;
+    std::string Trailer;
+    std::string TransferEncoding;
+    std::string Upgrade;
+    std::string Vary;
+    std::string Via;
+    std::string Warning;
+    std::string WWWAuthenticate;
+};
+class OperateData
+{
+public:
+    OperateData();
+    OperateData(OperateData &right);
+    RequestHeader requestHeader;
+    ResponseHeader responHeader;
+    std::unordered_map<std::string, std::string> body;
```

Comment:
request body? What about response?

Reply:
both header and body are store in format of unordered_map, user should choose to use as request data or response data

Reply:
How do they know which entry of header/body belongs to the response or the request? Do you want to separate them?

Reply:
One OperateData is only used as request data or response data. For example, if it used as request data, then the header in the OperateData only contain request header same as the body. So, the user should be aware of the naming of the OperateData to distinguish between request or response data.